gluten-substrait/src/main/scala/org/apache/gluten/execution/WriteFilesExecTransformer.scala

```diff
@@ -118,7 +193,7 @@ case class WriteFilesExecTransformer(
     }
 
     RelBuilder.makeWriteRel(
-      input,
+      inputRelNode,
```

Comment:
@dmsuehir After establishing the correct partition order in the pre-project , it seems necessary to also update the order of the `typeNodes`, `nameList `and `columnTypeNodes`.